src/common/lib/lnurl.ts

```diff
@@ -61,27 +68,51 @@ const lnurl = {
 
     return null;
   },
+
   async getDetails(lnurlString: string): Promise<LNURLError | LNURLDetails> {
     const url = normalizeLnurl(lnurlString);
-    let lnurlDetails = {} as LNURLDetails;
-    lnurlDetails.tag = url.searchParams.get("tag") as LNURLDetails["tag"];
-    if (lnurlDetails.tag === "login") {
-      lnurlDetails.k1 = url.searchParams.get("k1") || "";
-      lnurlDetails.action = url.searchParams.get("action") || undefined;
+    const searchParamsTag = url.searchParams.get("tag");
+
+    if (searchParamsTag && searchParamsTag === "login") {
```

Comment:
we have many such checks. there should be some way to validate an object against a schema without such nested `if`